service/rpc/availability.go

```diff
@@ -0,0 +1,44 @@
+package rpc
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+
+	"github.com/celestiaorg/celestia-node/service/share"
+)
+
+const heightAvailabilityEndpoint = "/data_available"
+
+func (h *Handler) handleHeightAvailabilityRequest(w http.ResponseWriter, r *http.Request) {
+	heightStr := mux.Vars(r)[heightKey]
+	height, err := strconv.Atoi(heightStr)
+	if err != nil {
+		writeError(w, http.StatusBadRequest, heightAvailabilityEndpoint, err)
+		return
+	}
+
+	header, err := h.header.GetByHeight(r.Context(), uint64(height))
+	if err != nil {
+		writeError(w, http.StatusInternalServerError, heightAvailabilityEndpoint, err)
+		return
+	}
+
+	err = h.share.Availability.SharesAvailable(r.Context(), header.DAH)
```

Comment:
Could this spawn too many sample requests via rpc?

Reply:
resolved?

Reply:
Poking @liamsi

Reply:
Closing as this is a non-issue.